checks/dangerous_workflow.go

```diff
@@ -0,0 +1,262 @@
+// Copyright 2021 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package checks
+
+import (
+	"fmt"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/ossf/scorecard/v3/checker"
+	sce "github.com/ossf/scorecard/v3/errors"
+)
+
+// CheckDangerousWorkflow is the exported name for Dangerous-Workflow check.
+const CheckDangerousWorkflow = "Dangerous-Workflow"
+
+//nolint:gochecknoinits
+func init() {
+	registerCheck(CheckDangerousWorkflow, DangerousWorkflow)
+}
+
+// Holds stateful data to pass thru callbacks.
+// Each field correpsonds to a dangerous GitHub workflow pattern, and
+// will hold true if the pattern is avoided, false otherwise.
+type patternCbData struct {
+	workflowPattern map[string]bool
+}
+
+// DangerousWorkflow runs Dangerous-Workflow check.
+func DangerousWorkflow(c *checker.CheckRequest) checker.CheckResult {
+	// data is shared across all GitHub workflows.
+	data := patternCbData{
+		workflowPattern: make(map[string]bool),
+	}
+	err := CheckFilesContent(".github/workflows/*", false,
+		c, validateGitHubActionWorkflowPatterns, &data)
+	return createResultForDangerousWorkflowPatterns(data, err)
+}
+
+// Check file content.
+func validateGitHubActionWorkflowPatterns(path string, content []byte, dl checker.DetailLogger,
```

Comment:
Hey Asra, some of the repos we check have random shell or python files in the .github/workflows directory. It would be good to check if this is a yaml file like in [pinned_dependencies.go](https://github.com/ossf/scorecard/blob/6562cc1f4488c7a018b5c6b4e031f990058d95a2/checks/pinned_dependencies.go#L525-L527). I'm creating a PR to fix this same issue in Token-Permissions.

Reply:
done!